### DISCUSSION TITLE
Class Components -> Functional Components: A Refactor

### DISCUSSION THREAD

**Discussion by matthewoestreich** [material-table-core/core#34] [Ideas]
General discussion around the refactoring of class components into functional components.

**Comment by matthewoestreich** [comment]
@Domino987 @MrWittman612 @naimbouallagui 

I think we need better commenting in this code base as well. What do you think about using this as an opportunity to comment important pieces of code? It isn't a difficult code base to understand, but comments def help speed up the process.

Thoughts?

**Reply by thefat32** [reply]
I agree

**Comment by thefat32** [comment]
I was about to start TS translation, but I think is better to refactor to functional components first
